uuidv7.zig

```diff
@@ -4,67 +4,56 @@
 // Licensed under the MPL-2.0
 
 const std = @import("std");
-const builtin = @import("builtin");
 
 pub fn main() !void {
     const stdout = std.io.getStdOut().writer();
-
-    const getrandom = if (builtin.os.tag == .windows) std.os.windows.RtlGenRandom else std.posix.getrandom;
-
-    const uuid7 = try generateUUIDv7(getrandom);
-    const uuid7str = try uuidToString(uuid7);
+    const uuid7 = try UUID7.generate();
+    const uuid7str = try uuid7.toString();
     try stdout.print("{s}\n", .{uuid7str});
 }
 
-pub const UUID = struct {
-    time_high: u32,
-    time_low: u16,
-    version_random1: u16,
-    variant_random2: u16,
-    random3: u16,
-    random4: u32,
+pub const UUID7 = packed struct {
+    // 48 bit big-endian unsigned number of Unix epoch timestamp as per Section 6.1.
+    unix_ts_ms: u48,
+
+    // 4 bit UUIDv7 version set as per Section 4
+    ver: u4 = 7,
+
+    // 12 bits pseudo-random data to provide uniqueness as per Section 6.2 and Section 6.6.
+    rand_a: u12,
+
+    // The 2 bit variant defined by Section 4.
+    @"var": u2 = 2,
+
+    // The final 62 bits of pseudo-random data to provide uniqueness as per Section 6.2 and Section 6.6.
+    rand_b: u62,
+
+    fn toString(self: @This()) ![36]u8 {
+        // time_h32-t_16-verh-varh-rand03rand04
+        // 019212d3-87f4-7d25-902e-b8d39fe07f08
+        const hex = std.fmt.hex(@as(u128, @bitCast(self)));
+        var buffer: [36]u8 = undefined;
+
+        _ = try std.fmt.bufPrint(&buffer, "{s}-{s}-{s}-{s}-{s}", .{
+            hex[0..][0..8].*,
```

Comment:
I've never seen syntax like this before.

As per our discussion, this would be the same:

```zig
hex[0..8].*,
hex[8..12].*,
hex[12..16].*,
hex[16..20].*,
hex[20..32].*,
```

The double indexing is to show the lengths. It could be done any number of times, albeit nonsensical:

```zig
hex[12..][0..4][0..4][0..4][0..4][0..4][0..4].*,
```

The `.*` is making a copy.

Reply:
Updated according to your preferences

Reply:
NO! Reverted to how you did it. I will learn at the feet of the master.